lightwood/encoder/text/pretrained.py

```diff
@@ -135,35 +80,43 @@ def __init__(
         self.uses_target = True
         self.output_size = None
 
-        # DEBUGGING!!!
         if self.embed_mode:
             log.info("Embedding mode on. [CLS] embedding dim output of encode()")
         else:
             log.info("Embedding mode off. Logits are output of encode()")
 
-    def prepare(self, train_priming_data: pd.Series, dev_priming_data: pd.Series, encoded_target_values: torch.Tensor):
+    def prepare(
+        self,
+        train_priming_data: pd.Series,
+        dev_priming_data: Optional[pd.Series],
+        encoded_target_values: torch.Tensor,
+    ):
         """
-        Prepare the encoder by training on the target.
+        Fine-tunes a transformer on the priming data.
 
-        Training data must be a dict with "targets" avail.
-        Automatically assumes this.
-        """
-        os.environ['TOKENIZERS_PARALLELISM'] = 'true'
-        priming_data = pd.concat([train_priming_data, dev_priming_data])
-        priming_data = priming_data.values
+        CURRENTLY WIP; train + dev are placeholders for a validation-based approach. 
+
+        Train + Dev are concatenated together and a transformer is then fine tuned with weight-decay applied on the transformer parameters. The option to freeze the underlying transformer and only train a linear layer exists if `frozen=True`. This trains faster, with the exception that the performance is often lower than fine-tuning on internal benchmarks.
+
+        :param train_priming_data: Text data in the train set
+        :param dev_priming_data: Text data in the dev set (not currently supported)
+        :param encoded_target_values: Encoded target labels in Nrows x N_output_dimension
+        """ # noqa
         if self.is_prepared:
             raise Exception("Encoder is already prepared.")
 
-        # TODO: Make tokenizer custom with partial function; feed custom->model
-        if self._tokenizer is None:
-            self._tokenizer = self._tokenizer_class.from_pretrained(self._pretrained_model_name)
+        os.environ['TOKENIZERS_PARALLELISM'] = 'true'
+
+        # TODO -> we shouldn't be concatenating these together
+        if dev_priming_data is not None:
```

Comment:
I don't think we should allow for null dev data, will make it a mess once we do try to implement early stopping or other things that require a dev set. Currently it's easy but won't be so in the future, + it changes the API from that which all other trainable encoders use (mandatory train and dev)

Reply:
we need to seriously revamp the way we use dev right now; early stopping is not a bulletproof solution, and if we dont provide a dev set, it seems to me that the more powerful strategies requiring dev simply can't run. This is as simple as asking `if dev is None -> pass`.

Reply:
So, my argument here is that giving an option where that option isn't needed is horrible and should never be done it.

1) Allows for unit tests to be unrepresentative of production use-cases
2) Asks maintainer to deal with more complexity in the future
3) Makes the API more complicated

While not adding anything.

`if dev is None -> pass` works when dev is not used for anything, once we start using dev what are we going to do to handle this edge case?

If you so desire I'm fine with removing the dev set from all encoders period, and then if they want to do early stopping they can split their own data. I think it would be a *horrible* decision, but it is better than this.

Reply:
How about passing an empty dev instead for cases where it's not needed? That way we keep the interface uniform and the check is now `if len(dev_priming_data) > 0:` or similar. I think the splitter is currently compatible with this as well.

Reply:
This is a good suggestion and a safe alternative IMO; if it doesn't work im happy to enforce a criteria with dev but right now it's strange we just concatenate the two together and do nothing with it.